src/extension.ts

```diff
@@ -127,13 +117,10 @@ export function activate(context: vscode.ExtensionContext) {
 		});
 	});
 
-	// Watch for active editor changes to update status bar visibility
-	context.subscriptions.push(disposable, statusBarItem);
-	context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(updateStatusBarVisibility));
-	context.subscriptions.push(vscode.window.onDidChangeTextEditorSelection(updateStatusBarVisibility));
+	context.subscriptions.push(disposable);
+	context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(updateEhrqlContext));
+	context.subscriptions.push(vscode.window.onDidChangeTextEditorSelection(updateEhrqlContext));
```

Comment:
This seems to cause the `updateEhrqlContext` to be called on every key press in a python file. That's probably fine because it runs very quickly. I assume it's here to cover the situation where you start a python file, and then at some point add the `ehrql` keyword. You could change it to `vscode.workspace.onDidSaveTextDocument` to just run it on file saves - though sometimes when I test it, it still triggers quite often as if vscode is secretly saving it.

Reply:
hmm, my intention was just for it to update when you activated it, on the assumption that most people will start from the research template and will already have an ehrql skeleton set up. Maybe `onDidChangeTextEditorSelection` is just not required